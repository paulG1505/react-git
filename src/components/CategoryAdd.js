import React,{useState} from 'react'
import PropTypes from 'prop-types'

export const CategoryAdd = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('')
    const handleInput=({target})=>{
        //en el target esta el valor
        setInputValue(target.value);
    }
    const handleSubmit=(e)=>{
        //prevenir el volver a cargar la pagina
        e.preventDefault()
        //para que no envie vacio
        if(inputValue.trim().length>1){

            //copia de lo que tiene el arreglo e incrementa primero
            setCategories(cat=>[inputValue,...cat])
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInput}
            />
        </form>
    )
}
//para que sea obigatorio el enviar
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
