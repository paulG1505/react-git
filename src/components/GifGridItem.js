import React from 'react'
import PropTypes from 'prop-types'
//impt snnipet

export const GifGridItem = ({url,title}) => {
    return (
        <div className="card animate__animated animate__backInRight">
            <img
                src={url} height='120px'
                // titulo alternativo
                alt={title}
            />
            <br/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}