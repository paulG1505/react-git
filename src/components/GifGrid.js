import React from 'react'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { loading, data } = useFetchGifs(category);
    return (
        <>
            <h3 className="animate__animated animate__fadeInDown" >{category}</h3>
            {loading ? 'Cargando...' :

                <div className="card-grid">

                    {/* al objeto lo descomponemos */}
                    {
                        data.map((img) => (
                            <GifGridItem
                                {...img}
                                key={img.id}
                            />
                        ))
                    }
                </div>
            }
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,

}