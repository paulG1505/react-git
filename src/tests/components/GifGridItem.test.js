import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import React from 'react'
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas a GridItem', () => {
    //renderizar el componente
    const wrapper=shallow(<GifGridItem title={'Hola'} url={'https://'} />)
    test('Debe mostrar el componente correctamente', () => {
        //comparamos con el snapshot
        //para actualizar un snapshot usar la tecla 'u'
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe tener un parrafo con el titulo', () => {
        //buscamos el parrafo
        const p=wrapper.find('p')
        //a este texto sin espacios lo comparamos con el titulo que mandamos
        expect(p.text().trim()).toBe('Hola')
    })
    test('Debe ser una imagen, tener url y tener titulo alternativo', () => {
        const img=wrapper.find('img')
        //analizar las propiedades img.props
        expect(img.prop('src')).toBe('https://')
        //comprueba el texto alternativo
        expect(img.prop('alt')).toBe('Hola')
    })
    test('Debe tener la animacion fadeIn', () => {
        const div=wrapper.find('div')
        const animated='animate__backInRight'
        //compromamos que tenga esa animacion siempre
        expect(div.prop('className').includes(animated)).toBe(true)
    })
    
})

