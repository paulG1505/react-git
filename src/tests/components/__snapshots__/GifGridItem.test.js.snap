// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__backInRight"
>
  <img
    alt="Hola"
    height="120px"
    src="https://"
  />
  <br />
  <p>
    Hola
  </p>
</div>
`;
